### ISSUE TITLE
Strange split of function

### ISSUE THREAD

**Issue by rfalke** [uxmal/reko#1073]
Version 0.10.0.0

Subject: https://github.com/rfalke/decompiler-subjects/tree/master/from_holdec/int_math_with_constants/arm64_elf_gcc_O0

```
// 0000000000000D28: void signed_char_add()
// Called from:
//      main
void signed_char_add()
{
        g_ptr1F2F88->b0000 = (byte) (int32) (int8) (uint32) (uint8) ((uint32) (uint8) (word32) g_ptr1F2F18->b0000 + 0x01);
        g_ptr1F2F88->b0001 = (byte) (int32) (int8) (uint32) (uint8) ((uint32) (uint8) (word32) g_ptr1F2F18->b0001 + 0x02);
        g_ptr1F2F88->b0002 = (byte) (int32) (int8) (uint32) (uint8) ((uint32) (uint8) (word32) g_ptr1F2F18->b0002 + 0x03);
...
        g_ptr1F2F88->b000F = (byte) (int32) (int8) (uint32) (uint8) ((uint32) (uint8) (word32) g_ptr1F2F18->b000F + 0x10);
        g_ptr1F2F88->b0010 = (byte) (int32) (int8) (uint32) (uint8) ((uint32) (uint8) (word32) g_ptr1F2F18->b0010 + 0x11);
        g_ptr1F2F88->b0011 = (byte) (int32) (int8) (uint32) (uint8) ((uint32) (uint8) (word32) g_ptr1F2F18->b0011 + 0x12);
        struct Eq_95 * x0_203 = g_ptr1F2F18;
        e843419@0005_0000003e_2d8(x0_203);
}

...
// 00000000001E172C: void e843419@0005_0000003e_32dc(Register (ptr64 Eq_110) x0, Register int32 w1)
// Called from:
//      e843419@0005_0000003e_2d8
void e843419@0005_0000003e_32dc(struct Eq_110 * x0, int32 w1)
{
        x0->b0145 = (byte) w1;
        g_ptr1F2F88->b0146 = (byte) (int32) (int8) (uint32) (uint8) ((uint32) (uint8) (word32) g_ptr1F2F18->b0146 + 0x47);
        g_ptr1F2F88->b0147 = (byte) (int32) (int8) (uint32) (uint8) ((uint32) (uint8) (word32) g_ptr1F2F18->b0147 + 0x48);
...
        g_ptr1F2F88->b01AA = (byte) (int32) (int8) (uint32) (uint8) ((uint32) (uint8) (word32) g_ptr1F2F18->b01AA - 0x55);
        g_ptr1F2F88->b01AB = (byte) (int32) (int8) (uint32) (uint8) ((uint32) (uint8) (word32) g_ptr1F2F18->b01AB - 0x54);
        struct Eq_95 * x0_1129 = g_ptr1F2F18;
        e843419@0005_0000003e_42d8(x0_1129);
}
...
```

**Comment by uxmal** [comment]
Your Ghidra sample exhibits the same issue. The issue is that there is a symbol at address `00000000001E172C`:
```
  186 e843419@0005_0000003e_32dc STT_FUNC         13 locl .text           00000000001E172C         8
```
It's stating that there is a function at that address (with a wonky name). The `b` instruction at address  
`0000000000001000` 14078217 is promoted to a Reko `call` instruction because it is jumping to a procedure. How does holdec handle this case?

**Comment by rfalke** [comment]
I suspected an issue with reko but it looks like the code generator (gcc 6.3.0) is responsible. Sorry for the noise.

I can not reproduce this with godbolt. And I'm not sure if I find the VM which I used initially.

It looks like gcc wanted to limited the function size. But the reason is unknown. The other subjects with different -O flag looking fine.

No idea what holdec will produce but probably something very similar to reko and ghidra.

So we have a compiler artifact which is IMHO not worth the effort to make the decompiler understand.
